src/main/java/org/sagebionetworks/bridge/hibernate/HibernatePermissionDao.java

```diff
@@ -0,0 +1,88 @@
+package org.sagebionetworks.bridge.hibernate;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+import com.google.common.collect.ImmutableSet;
+import org.sagebionetworks.bridge.dao.PermissionDao;
+import org.sagebionetworks.bridge.models.permissions.Permission;
+import org.sagebionetworks.bridge.models.permissions.EntityType;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Resource;
+
+import java.util.Set;
+
+@Component
+public class HibernatePermissionDao implements PermissionDao {
+    
+    static final String APP_ID = "appId";
+    static final String USER_ID = "userId";
+    static final String ENTITY_TYPE = "entityType";
+    static final String ENTITY_ID = "entityId";
+    
+    static final String GET_BY_USER = "FROM Permission WHERE appId=:appId AND userId=:userId";
+    static final String GET_BY_ENTITY_TYPE = "FROM Permission WHERE appId=:appId "+
+            "AND entityType=:entityType AND entityId=:entityId";
+    
+    private HibernateHelper hibernateHelper;
+    
+    @Resource(name = "mysqlHibernateHelper")
+    final void setHibernateHelper(HibernateHelper hibernateHelper) {
+        this.hibernateHelper = hibernateHelper;
+    }
+    
+    @Override
+    public Set<Permission> getPermissionsForUser(String appId, String userId) {
+        checkArgument(isNotBlank(appId));
+        checkArgument(isNotBlank(userId));
+        
+        QueryBuilder builder = new QueryBuilder();
+        builder.append(GET_BY_USER, APP_ID, appId, USER_ID, userId);
+        
+        return ImmutableSet.copyOf(hibernateHelper.queryGet(
+                builder.getQuery(), builder.getParameters(), null, null, Permission.class));
```

Comment:
Permissions are mutable. If this doesn't need to be a Set, I recommend using a List instead.

Reply:
Switched over to a List.